addons/website/static/src/interactions/carousel_slider.js

```diff
@@ -0,0 +1,55 @@
+import { registry } from "@web/core/registry";
+import { Interaction } from "@website/core/interaction";
+import { debounce } from "@web/core/utils/timing";
+
+export class CarouselSlider extends Interaction {
+    static selector = ".carousel";
+    dynamicContent = {
+        "img": {
+            "t-on-load": this.computeMaxHeight,
+        },
+        "_window": {
+            "t-on-resize": this.debouncedComputeMaxHeight,
+        },
+        ".carousel-item": {
+            "t-att-style": () => `min-height: ${this.maxHeight};`,
+        },
+    };
+    carouselOptions = undefined;
+
+    setup() {
+        this.debouncedComputeMaxHeight = debounce(() => this.computeMaxHeight(), 250);
```

Comment:
note that the debounce is a side effect => it should be cancelled on destroy, otherwise the code could be executed after destruction